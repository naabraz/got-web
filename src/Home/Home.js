import React, { useEffect } from 'react';

// @flow

import {
  Container,
  Title,
} from './Home.styles';

type Props = {
  withService: Function
}

export const Home = ({ withService }: Props) => {
  const callService = async () => {
    withService();
  };

  useEffect(() => {
    callService();
  }, []);

  return (
    <Container>
      <Title>
        Game of Thrones
      </Title>
    </Container>
  );
};
