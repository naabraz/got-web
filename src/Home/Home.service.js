import React from 'react';

export const getHouses = () => fetch('https://api.got.show/api/book/houses')
  .then(res => res.json())
  .then(data => data)
  .catch(e => new Error(e));

export const withService = Component => props => <Component withService={getHouses} {...props} />;
