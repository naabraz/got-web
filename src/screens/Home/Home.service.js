import React from 'react';
import { getHouses } from 'got-api-lib';

export const withService = Component => props => <Component withService={getHouses} {...props} />;
