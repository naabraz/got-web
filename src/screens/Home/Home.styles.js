import styled from 'styled-components';

export const Container = styled.div`
  display: flex
`;

export const Title = styled.h1`
  color: red
`;
