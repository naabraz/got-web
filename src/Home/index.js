import { withService } from './Home.service';
import { Home } from './Home';

export default withService(Home);
