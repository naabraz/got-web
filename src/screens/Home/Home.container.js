import React, { useEffect, useState } from 'react';
import { getHouses } from 'got-api-lib';

import { Home } from './Home';

export const HomeContainer = () => {
  const [houses, setHouses] = useState([]);

  useEffect(() => {
    const callService = async () => {
      const items = await getHouses();
      await setHouses(items);
    };

    callService();
  });

  return (
    <Home houses={houses} />
  );
};
