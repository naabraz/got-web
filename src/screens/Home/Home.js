// @flow
import React from 'react';

import { Container, Title } from './Home.styles';
import { GotList } from '../../components/GotList';

type Props = {
  houses: Array<Object>
}

export const Home = ({ houses }: Props) => (
  <Container>
    <Title>
      Game of Thrones
    </Title>
    <GotList houses={houses} />
  </Container>
);
