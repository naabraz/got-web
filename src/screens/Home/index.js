import { HomeContainer } from './Home.container';

export default HomeContainer;
