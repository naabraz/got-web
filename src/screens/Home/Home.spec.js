import React from 'react';
import renderer from 'react-test-renderer';
// import { mount } from 'enzyme';

import { Home } from './Home';

describe('Home screen', () => {
  it('Should match snapshot', () => {
    const houses = [{ name: 'Targaryen', id: 1 }, { name: 'Stark', id: 2 }];
    const HomeScreen = renderer.create(<Home houses={houses} />).toJSON();
    expect(HomeScreen).toMatchSnapshot();
  });

  // it('Should call API to get all houses', async () => {
  //   const houses = [{ name: 'Targaryen', id: 1 }, { name: 'Stark', id: 2 }];

  //   const component = mount(<Home houses={houses} />);

  //   setImmediate(() => {
  //     component.update();
  //     expect(props.withService).toHaveBeenCalled();
  //   });
  // });
});
