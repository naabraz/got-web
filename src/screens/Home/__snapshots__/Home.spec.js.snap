// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen Should match snapshot 1`] = `
<div
  className="sc-bdVaJa bvPUMP"
>
  <h1
    className="sc-bwzfXH kwYnvM"
  >
    Game of Thrones
  </h1>
  <ul
    className="sc-htpNat hHtpcc"
  >
    <li
      className="sc-bxivhb eBNJKg"
    >
      Targaryen
    </li>
    <li
      className="sc-bxivhb eBNJKg"
    >
      Stark
    </li>
  </ul>
</div>
`;
